### ISSUE TITLE
Include original log data of modified projects

### ISSUE THREAD

**Issue by elizlee** [isi-vista/curated-training-annotator#91]
As pointed out [here](https://github.com/isi-vista/curated-training-annotator/issues/86#issuecomment-664599434), whenever a new project file is imported to Inception, it appears to get a fresh `event.log` even if the project already has one. This affects the recorded annotation time stats and indicator lists of any modified project with annotations.

If we want to restore this lost information, since we have backups of each original log file, we could perhaps rename these to something like `event_1.log`, add them to each project's folder in the data repos, and have `ParseEventLogs` be able to read multiple log files if they exist in the given directory.

Suggestions on how to handle this are welcome.

**Comment by reckart** [comment]
When a project is imported into INCEpTION, the event log should be imported as well. And when you export it, the original (pre-import) events as well as and new events should be included in the freshly exported event log. Do you think you hit a but in INCEpTION?

**Comment by elizlee** [comment]
@reckart -- I used the remote API to import the modified projects. Would that make a difference?

**Comment by reckart** [comment]
I don't think so. Under the hood, project import/export via the remote API and the UI both end up in the same code.

**Comment by elizlee** [comment]
@reckart -- Do the imported projects need to have a certain degree of similarity to their originals so that the pre-import events are included in the new log - i.e. could the modified projects be different enough that they were interpreted as brand new projects? The imported project files were each given one or two new tags, and the original projects were imported by a different administrator using the UI.

Here is an example of an `event.log` which was part of the project ZIP when it was re-imported:
```
{"created":1590072001000,"event":"AfterProjectCreatedEvent","id":734821,"user":"liz_lee"}
{"annotator":"liz_lee","created":1590778680000,"details":{"query":"ill"},"event":"ExternalSearchQueryEvent","id":761228,"user":"liz_lee"}
{"created":1590778685000,"document_name":"NYT_ENG_20010921.0195","event":"AfterDocumentCreatedEvent","id":761229,"user":"liz_lee"}
{"created":1590778685000,"details":{"state":"NEW"},"event":"ProjectStateChangedEvent","id":761230,"user":"liz_lee"}
{"annotator":"liz_lee","created":1590778689000,"document_name":"NYT_ENG_20010921.0195","event":"AfterCasWrittenEvent","id":761231,"user":"liz_lee"}
{"created":1590778689000,"details":{"state":"ANNOTATION_INPROGRESS","previousState":"NEW"},"document_name":"NYT_ENG_20010921.0195","event":"DocumentStateChangedEvent","id":761232,"user":"liz_lee"}
{"created":1590778689000,"details":{"state":"ANNOTATION_INPROGRESS","previousState":"NEW"},"event":"ProjectStateChangedEvent","id":761233,"user":"liz_lee"}
{"annotator":"liz_lee","created":1590778689000,"details":{"state":"INPROGRESS","previousState":"NEW"},"document_name":"NYT_ENG_20010921.0195","event":"AnnotationStateChangeEvent","id":761234,"user":"liz_lee"}
{"annotator":"liz_lee","created":1590778689000,"document_name":"NYT_ENG_20010921.0195","event":"DocumentOpenedEvent","id":761235,"user":"liz_lee"}
{"annotator":"liz_lee","created":1590778705000,"document_name":"NYT_ENG_20010921.0195","event":"AfterCasWrittenEvent","id":761236,"user":"liz_lee"}
{"annotator":"liz_lee","created":1590778705000,"document_name":"NYT_ENG_20010921.0195","event":"DocumentOpenedEvent","id":761237,"user":"liz_lee"}
{"annotator":"liz_lee","created":1590778725000,"details":{"addr":23817,"begin":6166,"end":6169,"type":"webanno.custom.CTEventSpan","text":"ill"},"document_name":"NYT_ENG_20010921.0195","event":"SpanCreatedEvent","id":761238,"user":"liz_lee"}
.
.
.
```

This is the `event.log` from the same project that was exported after being re-imported:
```
{"created":1595368873000,"event":"AfterProjectCreatedEvent","id":916465,"user":"remote_admin"}
```

**Comment by reckart** [comment]
@UWinch @jcklie could you please check if we can reproduce this? Sounds like a bug and if it is we might still be able to get a fix for it into 0.17.0.

**Comment by UWinch** [comment]
Hi, could you please clarify some points for me:

- Which version of INCEpTION are you currently using?
- How does your work process in detail look like? My understanding is:

  1. Admin user imports project manually into inception
  2. A script exports the project from inception remotely (admin_remote user), adds some labels automatically and re-imports the project to Inception
  3. When exporting this project again manually or by API the event log has been overwritten

- Does the event log still exist with the previous entries after the exported project has been changed and re-zipped by the script?

Thanks

**Comment by elizlee** [comment]
@UWinch --
1. We are using INCEpTION 0.15.2
2. The process is
  i. Admin user imports project manually into inception
  ii. A script exports the project from inception remotely (admin_remote user)
  iii. A copy of the project is automatically generated, and this copy is modified by receiving some new labels for the tagset in the `exportedproject123.json` file
  iv. If steps 2 and 3 succeeded, the project on inception is automatically deleted - this is so that we don't end up with hundreds of `copy_of_` projects
  v. The modified copy is imported by the script
  vi. When exporting this project again manually or by API, the annotations themselves are still there, but the event log has been overwritten
3. Yes, the export and the modified copy still have the original event log with the previous entries.

**Comment by UWinch** [comment]
I ran your script `AddNewtags.kt`, the result .zip file did not have the subdirectories that exported Inception projects normally have (`sources`, `log`, `annotation` and `annotation_ser` etc.). After I added these manually, the log was imported/exported as expected. Could you check if your .zip has these subdirectories?

**Comment by elizlee** [comment]
@UWinch -- did you run `AddNewTags` on a project with annotations? If not, then it won't have those subdirectories, and we're not concerned about the logs from "new" projects. When I run it, both original and modified zips with annotations contain these:
* `annotation/` --> document directories --> `<annotator>.tsv`
* `annotation_ser/` --> document directories --> `INITIAL_CAS.ser`, `<annotator>.ser`
* `event.log`, with the Inception events recorded
* `exportedproject<string of numbers>.json`
* `source/` --> documents

Projects that have been curated also have `curation/` and `curation_ser/` subdirectories.

**Comment by UWinch** [comment]
Hi @elizlee 

sorry, that was my fault, the subdirectories are there. As far as I can see the zip files have entries that start with an extra slash, e.g. `/event.log`, `/source/`. Inception looks for a zip entry `event.log` and cannot find it. 

Could you try changing the places in your code that use 
`val inMemoryProjectPath = inMemoryFileSystem.getPath("/${projectFile.name}")` to 
`val inMemoryProjectPath = inMemoryFileSystem.getPath(projectFile.name)` to see if that helps?

**Comment by elizlee** [comment]
@UWinch -- The events are in the export now! Thank you so much for locating my bug!

**Comment by UWinch** [comment]
Great! Happy to help :)
